src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp

```diff
@@ -0,0 +1,601 @@
+    trustm_Open();
+    return_status = trustm_ecc_keygen(keyid, key_usage, OPTIGA_ECC_CURVE_NIST_P_256, pubkey, &pubKeyLen);
+
+    // Add signature length
+    VerifyOrExit(return_status == OPTIGA_LIB_SUCCESS, error = CHIP_ERROR_INTERNAL);
+
+    /* Set the public key */
+    VerifyOrReturnError((size_t) pubKeyLen > NIST256_HEADER_OFFSET, CHIP_ERROR_INTERNAL);
+    VerifyOrReturnError(((size_t) pubKeyLen - NIST256_HEADER_OFFSET) <= kP256_PublicKey_Length, CHIP_ERROR_INTERNAL);
+    memcpy((void *) Uint8::to_const_uchar(public_key), pubkey + NIST256_HEADER_OFFSET, pubKeyLen - NIST256_HEADER_OFFSET);
+
+    memcpy(&mKeypair.mBytes[0], trustm_magic_no, sizeof(trustm_magic_no));
+    mKeypair.mBytes[CRYPTO_KEYPAIR_KEYID_OFFSET] = (keyid >> (0 * 8)) & 0xFF;
+    mKeypair.mBytes[CRYPTO_KEYPAIR_KEYID_OFFSET + 1] = (keyid >> (1 * 8)) & 0xFF;
+
+    mInitialized = true;
+    error = CHIP_NO_ERROR;
+
+exit:
+    if (error != CHIP_NO_ERROR)
+    {
+        trustm_close();
+    }
+    return error;
+#endif
+}
+
+CHIP_ERROR P256Keypair::ECDSA_sign_msg(const uint8_t * msg, size_t msg_length, P256ECDSASignature & out_signature) const
+{
+#if !ENABLE_TRUSTM_GENERATE_EC_KEY
+    return ECDSA_sign_msg_H(&mKeypair, msg, msg_length, out_signature);
+#else
+    CHIP_ERROR error = CHIP_ERROR_INTERNAL;
+    optiga_lib_status_t return_status = OPTIGA_LIB_BUSY;
+
+    uint8_t signature_trustm[kMax_ECDSA_Signature_Length_Der] = { 0 };
+    uint16_t signature_trustm_len                             = (uint16_t) kMax_ECDSA_Signature_Length_Der;
+    uint8_t digest[32];
+    uint8_t digest_length = sizeof(digest);
+
+    memset(&digest[0], 0, sizeof(digest));
+    MutableByteSpan out_raw_sig_span(out_signature.Bytes(), out_signature.Capacity());
+
+    VerifyOrReturnError(msg != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(msg_length > 0, CHIP_ERROR_INVALID_ARGUMENT);
+    ChipLogDetail(Crypto, "ECDSA_sign_msg: Using TrustM for ecdsa sign!");
+    // Trust M Init
+    trustm_Open();
+    // Hash to get the digest
+    Hash_SHA256(msg, msg_length, &digest[0]);
+    uint16_t keyid =  (mKeypair.mBytes[CRYPTO_KEYPAIR_KEYID_OFFSET]) | (mKeypair.mBytes[CRYPTO_KEYPAIR_KEYID_OFFSET+1] << 8);
+    // Api call to calculate the signature
+    if (keyid == OPTIGA_KEY_ID_E0F2) {
+        return_status = trustm_ecdsa_sign(OPTIGA_KEY_ID_E0F2, digest, digest_length, signature_trustm, &signature_trustm_len);
+    } else {
+        return_status = trustm_ecdsa_sign(OPTIGA_KEY_ID_E0F0, digest, digest_length, signature_trustm, &signature_trustm_len);
+    }
+    
+    VerifyOrExit(return_status == OPTIGA_LIB_SUCCESS, error = CHIP_ERROR_INTERNAL);
+
+    error = EcdsaAsn1SignatureToRaw(kP256_FE_Length, ByteSpan{ signature_trustm, signature_trustm_len }, out_raw_sig_span);
+
+    // ChipLogError(NotSpecified, "EcdsaAsn1SignatureToRaw %" CHIP_ERROR_FORMAT, error.Format());
+
+    SuccessOrExit(error);
+
+    SuccessOrExit(out_signature.SetLength(2 * kP256_FE_Length));
```

Comment:
Linter complains about this and it seems correct: error is CHIP_NO_ERROR otherwise line above would have exited.

Reply:
Fixed these in the latest commits.